src/priorityqueue.jl

```diff
@@ -342,11 +348,66 @@ end
 
 # Unordered iteration through key value pairs in a PriorityQueue
 function _iterate(pq::PriorityQueue, state)
-    state == nothing && return nothing
     (k, idx), i = state
     return (pq.xs[idx], i)
 end
+_iterate(pq::PriorityQueue, ::Nothing) = nothing
 
 iterate(pq::PriorityQueue) = _iterate(pq, iterate(pq.index))
-
 iterate(pq::PriorityQueue, i) = _iterate(pq, iterate(pq.index, i))
+
+"""
+    PQOrderedIterator{K, V, O <: Ordering}
+
+Provides an ordered iterator interface to iterate a [`PriorityQueue`](@ref).
+
+```jldoctest
+
+julia> pq = PriorityQueue(["a" => 10, "b" => 5, "c" => 15])
+PriorityQueue{String,Int64,Base.Order.ForwardOrdering} with 3 entries:
+  "c" => 15
+  "b" => 5
+  "a" => 10
+
+julia> collect(pq)
+3-element Array{Pair{String,Int64},1}:
+ "c" => 15
+ "b" => 5 
+ "a" => 10
+
+julia> it = PQOrderedIterator(pq)
+PQOrderedIterator{String,Int64,Base.Order.ForwardOrdering}(PriorityQueue("c"=>15,"b"=>5,"a"=>10))
+
+julia> collect(it)
+3-element Array{Pair{String,Int64},1}:
+ "b" => 5 
+ "a" => 10
+ "c" => 15
+```
+"""
+mutable struct PQOrderedIterator{K, V, O <: Ordering}
```

Comment:
`PQOrderedIterator.pq` is never modified, so this can be immutable
```suggestion
struct PQOrderedIterator{K, V, O <: Ordering}
```